packages/bindgen/CMakeLists.txt

```
@@ -193,14 +194,38 @@ if(DEFINED CMAKE_JS_VERSION)
         OUTPUT node_init.cpp
         COMMAND ${NODE_FORCE_COLOR} ${CMAKE_CURRENT_SOURCE_DIR}/realm-bindgen.ts ${SPEC_ARGS} --template node --output ${CMAKE_CURRENT_BINARY_DIR}
         VERBATIM
-        MAIN_DEPENDENCY src/templates/node.ts
+        MAIN_DEPENDENCY src/templates/browser.ts
         DEPENDS
             ${SPEC_FILES}
             SpecJsonSchema
             ${TS_FILES}
     )
 
     target_sources(realm-js PRIVATE node_init.cpp ${CMAKE_JS_SRC} ../../src/node/platform.cpp)
+
+elseif(EMSCRIPTEN)    
+    include(browser.cmake)
+
+    add_custom_command(
+        OUTPUT browser_init.cpp
+        COMMAND ${NODE_FORCE_COLOR} ${CMAKE_CURRENT_SOURCE_DIR}/realm-bindgen.ts ${SPEC_ARGS} --template browser --output ${CMAKE_CURRENT_BINARY_DIR}
+        VERBATIM
+        MAIN_DEPENDENCY src/templates/browser.ts
+        DEPENDS
+            ${SPEC_FILE}
+            SpecJsonSchema
+            ${TS_FILES}
+    )
+
+    target_sources(realm-js PRIVATE browser_init.cpp ${CMAKE_JS_SRC} ../../src/browser/platform.cpp)
+
+    add_custom_command(
+        TARGET realm-js-wasm
+        COMMAND ${CMAKE_COMMAND} -E copy
+                ${CMAKE_CURRENT_BINARY_DIR}/realm-js-wasm.js ${CMAKE_CURRENT_BINARY_DIR}/realm-js-wasm.wasm
+                ${TYPESCRIPT_OUTPUT_DIR}        
+    )
```

```suggestion
    set_target_properties(realm-js-wasm PROPERTIES
        RUNTIME_OUTPUT_DIRECTORY ${TYPESCRIPT_OUTPUT_DIR}
    )
```
This should achieve the same effect, without hardcoding the output files explicitly. You might need to set `LIBRARY_OUTPUT_DIRECTORY` also, but I think `RUNTIME` should be sufficient.